### PR TITLE
ui-manchette: make the space time chart scale fit vertically

### LINKED ISSUE #313: GET+Manchette: can not see the whole path on proportional mode
In proportional mode, when the component is rendered for the first time, the whole path should be displayed.

The initial zoom level is not correct.

### PR DESCRIPTION
**Pull request by clarani**
closes #313

### CODE REVIEW

File: ui-manchette/src/components/Manchette.tsx [hunk 1]
```diff
@@ -59,7 +64,7 @@ const Manchette: FC<ManchetteProps> = ({
     yOffset: 0,
     panning: null,
     scrollPosition: 0,
-    isProportional: false,
+    isProportional: true,
```
Comment (SharglutDev) [inline ui-manchette/src/components/Manchette.tsx:67]: We want to be by default in proportional ? Or only the story should be like that ?

File: ui-manchette/src/components/helpers.ts [hunk 2]
```diff
@@ -55,23 +55,33 @@ export const calcOperationalPointsToDisplay = (
...
   operationalPoints: StyledOperationalPointType[],
+  height: number,
   zoom: number,
   isProportional: boolean
-) =>
-  operationalPoints.map((op, index) => {
+) => {
+  if (operationalPoints.length < 2) return [];
+  const from = operationalPoints[0].position;
+  const to = operationalPoints[operationalPoints.length - 1].position;
```
Comment (SharglutDev) [inline ui-manchette/src/components/helpers.ts]: How would do you feel to start using this syntax ? I think it's more concise. And it forces us to check if the value is undefined so maybe it's more safe ?

```suggestion
  const to = operationalPoints.at(-1)!.position;
```

File: ui-manchette/src/components/helpers.ts
Comment (SharglutDev) [inline ui-manchette/src/components/helpers.ts:1]: If I test with a lot of ops, I get this in proportional. Maybe something should also be changed in `calcOperationalPointsToDisplay` to prevent overlapping ?

Capture d’écran 2024-08-29 à 10 28 27

File: ui-manchette/src/components/helpers.ts [hunk 3]
```diff
@@ -83,13 +93,28 @@ export const getOperationalPointsWithPosition = (
...
-  return scales;
+export const getScales = (
+  ops: OperationalPoint[],
+  height: number,
+  isProportional: boolean,
+  zoomY: number
+) => {
+  if (ops.length < 2) return [];
+  const from = ops[0].position;
+  const to = ops[ops.length - 1].position;
```
Comment (SharglutDev) [inline ui-manchette/src/components/helpers.ts]: Same here for `.at(-1)` if you decide to use it.

File: ui-manchette/src/components/helpers.ts [hunk 4]
```diff
@@ -83,13 +93,28 @@ export const getOperationalPointsWithPosition = (
...
+
+  const totalDistance = to - from;
+  const heightWithoutFinalOp = height - FOOTER_HEIGHT - OP_LINE_HEIGHT / 2;
```
Comment (SharglutDev) [inline ui-manchette/src/components/helpers.ts:114]: We use this two times in this file. Should we extract this logic in a function with a param for the last part ? OR do you think it's overkill ?

```suggestion
  const heightWithoutFinalOp = getHeightWithoutFinalOp(OP_LINE_HEIGHT / 2);
```

File: ui-manchette/src/components/Manchette.tsx
Comment (SharglutDev) [inline ui-manchette/src/components/Manchette.tsx:1]: Capture d’écran 2024-08-29 à 10 28 27

Doesn't seem to be related to your PR, but if I understand correctly, all the horizontal blue lines we see on the space time chart correspond to ops ? If the correspond ops are hidden, should we still see these lines ?

Also the alignment between an op and the blue line doesn't seem to be very good.

**Review by SharglutDev — CHANGES_REQUESTED** [review]
Thank you for this, left some comments.

File: ui-manchette/src/components/Manchette.tsx (hunk 1 above)
Comment (clarani) [reply ui-manchette/src/components/Manchette.tsx:67]: We want it by default in proportional (asked by Loïc)

File: ui-manchette/src/components/helpers.ts (hunk 2 above)
Comment (clarani) [reply ui-manchette/src/components/helpers.ts]: Yes you're right, it might be better :+1:

File: ui-manchette/src/components/helpers.ts (hunk 3 above)
Comment (clarani) [reply ui-manchette/src/components/helpers.ts]: Done

File: ui-manchette/src/components/Manchette.tsx
Comment (clarani) [reply ui-manchette/src/components/Manchette.tsx:1]: Yes it's not related to this PR, so I leave it for another PR :)

File: ui-manchette/src/components/helpers.ts
Comment (clarani) [reply ui-manchette/src/components/helpers.ts:1]: Good catch, I fixed !

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested, good job ! 

We should just create an issue for the alignment problem.

**Comment by clarani** [comment]
Here is the ticket for the not aligned PR: https://github.com/OpenRailAssociation/osrd-ui/issues/396
